cip/1.accepted/CIP2018-10-19-Delete-Semantics.adoc

```diff
@@ -0,0 +1,49 @@
+= CIP2018-10-19 Semantics of Deleted Elements
+:numbered:
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Author:* Tobias Lindaaker <tobias.lindaaker@neotechnology.com>
+
+toc::[]
+
+== Semantics of Deleted Elements
+
+Cypher allows read statements to occur after update statements.
+This includes reading statements after statements that delete elements.
+Since the driving table of the preceding query is retained into the succeeding reading statements, this means that entries in the driving table that previously contained elements might now contain elements that have been deleted.
+
+The semantics of accessing such elements is the same as accessing `NULL` values.
+
+This can be thought of as replacing all occurrences of the deleted elements (anywhere) in the driving table (including in nested values) with `NULL`, or as treating the deleted elements as _effectively_ `NULL`.
+
+These semantics effect the `DELETE` statement itself, even if not succeeded by further read statements, since the same element can occur in multiple rows in the driving table.
+
+
+=== Accessing properties of deleted Elements
+
+Accessing properties of deleted elements produces a `NULL` value, just like accessing a property from a `NULL` value would.
+
+=== Pattern matching on deleted Elements
+
+Pattern matching on deleted elements produces a single row containing only `NULL` bindings, in the same way as `OPTIONAL MATCH` would.
+
+=== Deleting deleted Elements
+
+Deleting a deleted element (or any `NULL` value) is a no-op.
+
+=== Equality of deleted Elements
```

Comment:
This is a change from current behaviour, where two references to the same element are considered equal even after the element has been deleted.